### PR TITLE
Full Site Editing: confirm changes to template parts

### LINKED ISSUE #33629: Full Site Editing: header changes pre-publish confirmation [3]
Before the header changes have been published/updated, a pre-publish confirmation should be shown:

Screenshot 2019-06-05 at 13 43 16

Given the context this flow should be applied when editing `wp_template` and `wp_template_part` CPTs in general.

### PR DESCRIPTION
**Pull request by mattwiebe**
#### Changes proposed in this Pull Request

* Add a post-update confirmation when updating a template part

in-action

#### Testing instructions

* load an existing Template Part post
* update it
* ensure that a confirmation panel appears after clicking the button
* since the button is hacked into place atop the "real" update button, try it at various screen sizes, devices, etc

Fixes #33629

### CODE REVIEW

**Comment by vindl** [comment]
Nice workaround @mattwiebe!

I noticed a visual glitch when switching between normal and fullscreen mode:

1. Open an existing template part in normal mode. 
2. Switch to fullscreen mode.
3. Add some new text to the title.

Screenshot 2019-06-14 at 15 56 14

Something similar can be observed when toggling back to normal mode:

Screenshot 2019-06-14 at 15 55 40

However, since we only target the release on Dotcom now, this issue shouldn't be a problem because we enforce fullscreen mode (at least for Calypso). Having said that, I wouldn't consider it a blocker - it should be good to land when `package-lock` is removed.

File: apps/full-site-editing/package-lock.json [hunk 1]
```diff
@@ -0,0 +1,2111 @@
+{
```
[the remark this answers is not in the document]
Comment (marekhrabe) [reply apps/full-site-editing/package-lock.json]: up to y'all to make the decision but it's mean to be committed as per docs https://docs.npmjs.com/files/package-lock.json

**Comment by mattwiebe** [comment]
I'll see if I can find a quick fix for the fullscreen issue and also seems like the `package-lock.json` should stay

**Comment by mattwiebe** [comment]
Okay fullscreen position issue handled, this should be good to go

**Comment by rodrigoi** [comment]
I think we are overengineering the problem a bit. I know that the extension points to implement something like what Figma describes are not there, but I think we can do something a lot simpler using `PluginPrePublishPanel` and some notices.

For example, the code below disables _Publish_/_Update_ with a Pre-Publish Panel or a Notice, and won't let the User save changes until acknowledging the danger of changing the _Template Part_

```jsx
import { __ } from '@wordpress/i18n';
import { compose } from '@wordpress/compose';
import { PluginPrePublishPanel } from '@wordpress/edit-post';
import { registerPlugin } from '@wordpress/plugins';
import { Button } from '@wordpress/components';
import { withDispatch, withSelect } from '@wordpress/data';
import { useEffect, useState } from '@wordpress/element';

const TemplatePartPrePublishPanel = compose(
	withDispatch( dispatch => {
		const { removeEditorPanel } = dispatch( 'core/edit-post' );
		const { lockPostSaving, unlockPostSaving } = dispatch( 'core/editor' )
		const { createWarningNotice } = dispatch ( 'core/notices' );

		return { createWarningNotice, lockPostSaving, unlockPostSaving };
	} ),
)( ( { createWarningNotice, lockPostSaving, unlockPostSaving } ) => {
	const [ showNotice, setShowNotice ] = useState( true );
	const message = __( 'Changes you publish will update all pages with this template part.' );

	useEffect( () => {
		if ( ! showNotice ) {
			return;
		}

		createWarningNotice( message, { isDismissible: false, actions: [
			{
				label: __( 'Agreed' ),
				onClick: () => unlockPostSaving()
			}
		] } );
		setShowNotice( false );
		lockPostSaving();
	} );

	return (
		<PluginPrePublishPanel
			title={ __( 'Publish a Template Part' ) }
			initialOpen={ true }
		>
			<p>{ message }</p>
			<Button onClick={ () => unlockPostSaving() } isDefault>{ __( 'Agreed' ) }</Button>
		</PluginPrePublishPanel>
	);
}
);

if ( 'wp_template_part' === fullSiteEditing.editorPostType ) {
	registerPlugin( 'fse-template-part-pre-publish-panel', {
		render: TemplatePartPrePublishPanel,
	} );
}
```
FireShot Capture 014 - Add New Template Part ‹ Gutenberg Dev — WordPress - localhost

It's not ideal, and it's not exactly what's designed, but we won't be duplicating a lot of code that we'll have to maintain later. And could give us a starting point for adding the features we need to core.

**Comment by vindl** [comment]
> I think we are overengineering the problem a bit. I know that the extension points to implement something like what Figma describes are not there, but I think we can do something a lot simpler using PluginPrePublishPanel and some notices.

This implementation looks cleaner but we'll probably have to run it by design first. I wouldn't mind switching to it.

File: apps/full-site-editing/package-lock.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (vindl) [reply apps/full-site-editing/package-lock.json]: My question was mostly due to the fact that we don't seem to include it as part of the other similar apps and packages inside the monorepo.

**Comment by mattwiebe** [comment]
Thanks for the alternative thinking @rodrigoi - it's something good to fall back on if this implementation winds up becoming a problem.

**Pull request merged**